projects/components/src/lib/directives/show-clipped-text.directive.ts

```diff
@@ -0,0 +1,278 @@
+     *     TL -> BR        TR -> BL
+     *     BL -> TR        BR -> TL
+     * @param rect THe dimensions of the clipped element
+     */
+    calculatePosition(rect: ClientRect): TooltipPosition {
+        const { innerWidth, innerHeight } = window;
+        const rectCenter = { x: rect.left + rect.width / 2, y: rect.top + rect.height / 2 };
+
+        // It's on the right
+        if (rectCenter.x > innerWidth / 2) {
+            if (rectCenter.y > innerHeight / 2) {
+                // Bottom right quad
+                return TooltipPosition.tl;
+            } else {
+                // Top right quad
+                return TooltipPosition.bl;
+            }
+        }
+
+        // It's on the left
+        if (rectCenter.y > innerHeight / 2) {
+            // Bottom left quad
+            return TooltipPosition.tr;
+        } else {
+            // Top left quad
+            return TooltipPosition.br;
+        }
+    },
+};
+
+type EventHandler = (e: MouseEvent) => void;
+
+function watchEvents(el: HTMLElement, mouseIn: EventHandler, mouseOut: EventHandler): void {
+    el.addEventListener('mouseenter', mouseIn);
+    el.addEventListener('mouseleave', mouseOut);
+}
+
+function unwatchEvents(el: HTMLElement, mouseIn: EventHandler, mouseOut: EventHandler): void {
+    el.removeEventListener('mouseenter', mouseIn);
+    el.removeEventListener('mouseleave', mouseOut);
+}
+
+function setStyle(el: HTMLElement, style: Partial<CSSStyleDeclaration>): void {
+    Object.assign(el.style, style);
+}
+
+/**
+ * Displays a clarity tooltip with the full contents of a host element on hover but only if
+ * the elements is clipped.
+ */
+@Directive({
+    selector: '[vcdShowClippedText]',
+})
+export class ShowClippedTextDirective implements OnDestroy, OnInit {
+    /** To destroy the tooltip when no longer needed */
+    static instanceCount = 0;
+
+    /** Size of tooltip */
+    @Input('vcdShowClippedText') tooltipSize = TooltipSize.md;
+
+    // tslint:disable-next-line:no-input-rename
+    @Input('vcdShowClippedTextMouseOutDelay')
+    mouseoutDelay = 500;
+
+    /**
+     * The HTML element receiving the directive
+     */
+    public hostElement: HTMLElement = this.host.nativeElement;
+
+    /**
+     * Be notified whenever the host element changes content or its CSS style
+     */
+    private mutationObserver = new MutationObserver(() => {
+        // Make sure isMouseOver is first. It's an optimization to avoid measuring the DOM
+        // Also don't update the tooltip if content changes but the mouse is over a different host
+        if (tip.isMouseOver && this.hostElement === tip.currentHost) {
+            if (this.isOverflowing()) {
+                tip.update();
+            } else {
+                tip.hideTooltip(this.mouseoutDelay);
+            }
+        }
+    });
+
+    constructor(private host: ElementRef) {}
+
+    ngOnInit(): void {
+        ShowClippedTextDirective.instanceCount++;
+        tip.create();
```

Comment:
Can the tip singleton support multiple instances/why keep track of instanceCount?

Reply:
So we can remove it after the final directive is destroyed